### ISSUE TITLE
Interproscan get stuck in "90% completed"

### ISSUE THREAD

**Issue by NextGenBioinformatics** [ebi-pf-team/interproscan#28]
Hi,

Whenever I use interproscan on about 10000 protein sequences, it will get stuck in "90% completed". I provided 16 cores for this job and used the default interproscan.properties file.
----------------------------------------------------------
22/04/2017 16:16:51:838 Running the following analyses:
[CDD-3.14,Coils-2.2.1,Gene3D-4.1.0,Hamap-201701.18,MobiDBLite-1.0,Pfam-30.0,PIRSF-3.01,PRINTS-42.0,ProDom-2006.1,ProSitePatterns-2
0.132,ProSiteProfiles-20.132,SFLD-2,SMART-7.1,SUPERFAMILY-1.75,TIGRFAM-15.0]
Available matches will be retrieved from the pre-calculated match lookup service.

Matches for any sequences that are not represented in the lookup service will be calculated locally.
22/04/2017 16:47:05:527 25% completed
22/04/2017 17:03:46:760 51% completed
22/04/2017 17:11:33:174 75% completed
22/04/2017 17:22:40:969 90% completed
----------------------------------------------------------

Could you please help me out about this?

Many thanks,

Regards,

Tom

**Comment by mifraser** [comment]
Hi Tom,

The latest Gene3D 4.1.0 post processing step seems to require more memory for some protein sequences than in past releases. 

We'll look into what to do for the next release, the quick fix for now would be to increase the "-Xmx2048M" max memory value in your interproscan.sh file (e.g. double it). Or you can try running with smaller chunks of input sequences.

Hope that helps?

Matt

**Comment by wangdp123** [comment]
Hi Matt,

I have tried to increase the maximum memory in the interproscan.sh file like this:

worker.command=java -Xms32m -Xmx4096m -jar interproscan-5.jar
worker.high.memory.command=java -Xms32m -Xmx4096m -jar interproscan-5.jar

Unfortunately, the issue remains. 

I can't pass the "90% completed". Would you like to help me further?

Thank you,

Tom

**Comment by mifraser** [comment]
Hi Tom,

That looks like you are editing the max memory of the workers in the interproscan.properties file. But I meant the max memory of the top level interproscan.sh file. You may need to double or triple the number! We'll look into increasing this default for the next release.

Hope that helps?

Matt

**Comment by wangdp123** [comment]
Hi Matt,

I have tripled the number in the interproscan.sh. It seems to pass the 90% and reach the 100%.

Unfortunately, there are NOT any final resultant files such as *.tsv, *.gff3, and *.html generated and the program is still running and never stopped after reaching 100%. Besides, I checked the jobWriteOutput directory in temporary directory and it is empty and the temporary directory still exists.

---------------------------------------------------------------------------------------------

02/05/2017 13:17:59:140 Welcome to InterProScan-5.23-62.0
02/05/2017 13:18:12:682 Running InterProScan v5 in STANDALONE mode... on Linux
02/05/2017 13:18:21:147 Loading file pep.fa
02/05/2017 13:18:21:166 Running the following analyses:
[CDD-3.14,Coils-2.2.1,Gene3D-4.1.0,Hamap-201701.18,MobiDBLite-1.0,PANTHER-11.1,Pfam-30.0,PIRSF-3.01,PRINTS-42.0,ProDom-2006.1,ProS
itePatterns-20.132,ProSiteProfiles-20.132,SFLD-2,SMART-7.1,SUPERFAMILY-1.75,TIGRFAM-15.0]
Available matches will be retrieved from the pre-calculated match lookup service.

Matches for any sequences that are not represented in the lookup service will be calculated locally.
02/05/2017 13:24:47:592 Uploaded/Stored 30000 sequences for analysis
02/05/2017 18:07:15:755 25% completed
02/05/2017 20:58:56:966 50% completed
02/05/2017 21:43:19:412 75% completed
02/05/2017 22:59:07:733 90% completed
03/05/2017 00:20:56:437 100% completed

---------------------------------------------------------------------------------------------

Any thoughts?

Thank you,

Tom

**Comment by mifraser** [comment]
Hi Tom,

Glad you made some progress. Did it ever finish? The very last thing should say "100% done:  InterProScan analyses completed" so it was still running. I agree the "100% completed" message is confusing! We'll look into the number rounding for future releases, that probably means it is 99.5% complete with just the output writing step remaining.

Unfortunately with 30,000 sequence results to retrieve and write for all default applications that final output step could take a long time. The performance will be much better if you split your input into chunks of smaller numbers of sequences (e.g. 3 x 10,000 or less) and then put the results together afterwards. 

Kind regards,

Matt

**Comment by gabo-ing** [comment]
Hi, I've got the exact same problem, however I splitted the original fasta into chunks of ~300 nucleotide sequences. However, it does not finish and still gets stuck at 90%, even with a fasta file of 242 sequences. 
I am trying the suggestion of increasing Xmx so let's see what happens

**Comment by gtduarte** [comment]
Hello,

I had the same issue running the standalone mode on two different machines, one with 8 GB and the other one with 32GB (both running the latest release 5.26-65.0 on Ubuntu 16.04). 

Since I want to run the analysis on my transcriptome assembly, I divided it into fasta files with 1000 nucleotide sequences each. Then I run a test using the first fasta file on the 8GB machine, and it completed the analysis in 12h. So then I started the analysis on both machines with two other 1000 sequences, but both got stuck at 90% (with only java running at the end but no progress nor error reports). The only difference that I noticed between the run that worked and those two that did not was that the program reported a higher number of unique sequences uploaded/stored for analysis when it worked (51731) than when it didn't (42429 and 34627), although I don't know if it is related to the issue.

Nevertheless, I stopped both runs after 24h without progress and now I am starting it again with files with 500 sequences.... hope it works.

Kind regards
